src/services/services.ts

```diff
@@ -2599,24 +2585,18 @@ module ts {
                     /// TODO filter meaning based on the current context
                     let scopeNode = getScopeNode(previousToken, position, sourceFile);
                     let symbolMeanings = SymbolFlags.Type | SymbolFlags.Value | SymbolFlags.Namespace | SymbolFlags.Alias;
-                    let symbols = typeInfoResolver.getSymbolsInScope(scopeNode, symbolMeanings);
-
-                    getCompletionEntriesFromSymbols(symbols, activeCompletionSession);
+                    
+                    // Filter down to the symbol that matches the symbolName if we were given one.
+                    let predicate: (s: Symbol) => boolean = symbolName 
+                        ? s => getValidCompletionEntryDisplayName(s, target) === symbolName
+                        : undefined;

```

Comment:
Do you need the full type annotation for `predicate`? If you annotate `s`, I don't think you get `any`, do you?